src/Bicep.Cli.IntegrationTests/DecompileParamsCommandTests.cs

```diff
@@ -96,7 +96,7 @@ public async Task Decompile_ValidParamFileWithBicepPath_ShouldSucceed()
   }
 }";
             var expectedOutput =
-@"using './dir/main.bicep'
+@"using 'dir/main.bicep'
```

Comment:
Can you explain the reason for all these changes?

Reply:
I noticed that the `SourceAwareJsonTextWriter` class takes a dependency on `IFileResolver`, purely to use the `GetRelativePath` method ([here](https://github.com/Azure/bicep/blob/03d846a065cbf5e63e88380d8889d38f65512dc2/src/Bicep.Core/Emit/PositionTrackingJsonTextWriter.cs#L245C1-L245C117)).

1. Conceptually it felt odd that generating a json file **should** have a dependency on the file system.
1. I noticed that we were actually converting from a `Uri` to a `string`, purely to call this method - whereas we already have capabilities to determine a relative URI path without having to do the conversion. Converting between string<->URI is tricky and can be error-prone, and isn't technically necessary.

To preserve existing behavior for other use-cases, I changed the behavior of `GetRelativePath` [here](https://github.com/Azure/bicep/blob/03d846a065cbf5e63e88380d8889d38f65512dc2/src/Bicep.Core/FileSystem/PathHelper.cs#L221C30-L221C30) to not explicitly add in the `./`. I considered making this customizable, but I also have started to prefer the `'main.bicep'` rather than `'./main.bicep'` convention, so I left like this.